src/material/form-field/form-field.ts

```diff
@@ -652,41 +639,43 @@ export class MatFormField
    * not need to do this because they use a fixed width for prefixes. Hence, they can simply
    * incorporate the horizontal offset into their default text-field styles.
    */
-  private _updateOutlineLabelOffset() {
-    if (!this._platform.isBrowser || !this._hasOutline() || !this._floatingLabel) {
-      return;
-    }
-    const floatingLabel = this._floatingLabel.element;
-    // If no prefix is displayed, reset the outline label offset from potential
-    // previous label offset updates.
-    if (!(this._iconPrefixContainer || this._textPrefixContainer)) {
-      floatingLabel.style.transform = '';
-      return;
-    }
-    // If the form field is not attached to the DOM yet (e.g. in a tab), we defer
-    // the label offset update until the zone stabilizes.
-    if (!this._isAttachedToDom()) {
-      this._needsOutlineLabelOffsetUpdateOnStable = true;
-      return;
-    }
-    const iconPrefixContainer = this._iconPrefixContainer?.nativeElement;
-    const textPrefixContainer = this._textPrefixContainer?.nativeElement;
-    const iconPrefixContainerWidth = iconPrefixContainer?.getBoundingClientRect().width ?? 0;
-    const textPrefixContainerWidth = textPrefixContainer?.getBoundingClientRect().width ?? 0;
-    // If the directionality is RTL, the x-axis transform needs to be inverted. This
-    // is because `transformX` does not change based on the page directionality.
-    const negate = this._dir.value === 'rtl' ? '-1' : '1';
-    const prefixWidth = `${iconPrefixContainerWidth + textPrefixContainerWidth}px`;
-    const labelOffset = `var(--mat-mdc-form-field-label-offset-x, 0px)`;
-    const labelHorizontalOffset = `calc(${negate} * (${prefixWidth} + ${labelOffset}))`;
-
-    // Update the translateX of the floating label to account for the prefix container,
-    // but allow the CSS to override this setting via a CSS variable when the label is
-    // floating.
-    floatingLabel.style.transform = `var(
-        --mat-mdc-form-field-label-transform,
-        ${FLOATING_LABEL_DEFAULT_DOCKED_TRANSFORM} translateX(${labelHorizontalOffset})
-    )`;
+  private _updateOutlineLabelOffsetAfterNextRender() {
+    afterNextRender(() => {
+      if (!this._platform.isBrowser || !this._hasOutline() || !this._floatingLabel) {
```

Comment:
nit: the `isBrowser` check should not be required anymore (the `afterNextRender` is only invoked on the client atm):
```suggestion
      if (!this._hasOutline() || !this._floatingLabel) {
```